### PR TITLE
Lab 4

### PR DESCRIPTION
**Pull request by Andrew174194**
Pull request for lab 4
by Andrey Khoroshavin (kill team, CS-01 academic group)

For comparison between Lab 3 and Lab 4

### CODE REVIEW

File: terraform/GitHub/variables.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+variable "token" {
+  type        = string
+  description = "Specifies the GitHub PAT token or `GITHUB_TOKEN`"
+  sensitive   = true
```
Comment (ntdesmond) [inline terraform/GitHub/variables.tf:4]: good one

File: terraform/TF.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+}
+```
+
+```terraform state list```
+```sh
+    docker_container.msc_time
+```
+
+## AWS infrastructure
+> Sorry, we need to pay to use AWS but we live in Russia. I can use VPN (and what I do), but I have no foreign bank card to link with account...
```
Comment (ntdesmond) [inline terraform/TF.md:77]: Yandex Cloud then?...

**Review by ntdesmond — COMMENTED** [review]
cloud where

File: terraform/Docker/main.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  required_providers {
+    docker = {
+      source  = "kreuzwerker/docker"
+      version = ">= 2.13.0"
+    }
+  }
+}
+
+provider "docker" {
+  host = "npipe:////.//pipe//docker_engine"
```
Comment (DOCtorActoAntohich) [inline terraform/Docker/main.tf:11]: windows moment :trollface:

File: terraform/TF.md (hunk 2 above)
Comment (DOCtorActoAntohich) [reply terraform/TF.md:77]: amongus then

**Review by DOCtorActoAntohich — APPROVED** [review]
rip

File: terraform/TF.md (hunk 2 above)
Comment (Andrew174194) [reply terraform/TF.md:77]: Now it is here. Thanks for alternative

File: terraform/Yandex Cloud/main.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  required_providers {
+    yandex = {
+      source = "yandex-cloud/yandex"
+    }
+  }
+  required_version = ">= 0.13"
+}
+
+provider "yandex" {
+  # Don't forget to set YC_TOKEN var for authentication
```
Comment (DOCtorActoAntohich) [inline terraform/Yandex Cloud/main.tf:11]: instead, you might make an uninitialized variable in `*.tf` and initialize it using `*.tfvars` file

**Review by DOCtorActoAntohich — APPROVED** [review]
gud job now :trollface:

**Pull request closed without merging**